### DISCUSSION TITLE
How stable is ntex?

### DISCUSSION THREAD

**Discussion by zmlzm** [ntex-rs/ntex#43] [General]
Hi, The ntex's api doc show it's a good framework for web things, I have a new project and want to try it. 
But i don't how stable is ntex? Is there a time for 1.0 or somthings need be in to 1.0,
Is there a roadmap for it. thanks

**Comment by zzau13** [comment]
I use it when they let me choose technologies. 
There are many others that are used much more, less stable. By the code that I have reviewed.

**Comment by fafhrd91** [comment]
i consider ntex stable. i use it in prod, for my work. i dont have specific roadmap, but i am planing to introduce uring based runtime and make tokio fully optional.

**Reply by zmlzm** [reply]
oh it is make sense，now days I often read that io_using will change the programming way.  it is happy to know this.

**Reply by fakeshadow** [reply]
> i consider ntex stable. i use it in prod, for my work. i dont have specific roadmap, but i am planing to introduce uring based runtime and make tokio fully optional.

That's really nice. It would be a big step up on handling IO. So it would be uring for linux and tokio for macos and windows right?

**Reply by fafhrd91** [reply]
i’d like to create custom reactor based on mio and keep tokio only for compatibility with ecosystem

**Reply by fakeshadow** [reply]
> i’d like to create custom reactor based on mio and keep tokio only for compatibility with ecosystem

That's really cool. ntex would be more or less a complete network stack with it.

**Comment by clia** [comment]
Great jobs! Wait to see if it could reduce the kernel CPU consumptions.
